src/w2grid.js

```diff
@@ -498,6 +498,7 @@ class w2grid extends w2base {
         } else {
             this.refresh()
         }
+        this.trigger("add", {target: this.name})
```

Comment:
I think it would be useful to pass on `record` and `first` variables with the event, as those are arguments in the functio. Also, add spacing after `{` and before `}`

Also, I noticed that there is already an add event (search the code) and it is when add button is clicked, so event name should be `addRecords`,

Also, you should have event's before and after loop. See how other events are triggered, with edata.finish(). Event should be cancellable too.